### PR TITLE
neopop-sdl: update `head` URL

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
For #88329

### CODE REVIEW

**Comment by carlocab** [comment]
```
curl: (35) error:06FFF089:digital envelope routines:CRYPTO_internal:bad key length
```
Probably needs `using: :homebrew_curl`. That said, this isn't bottled yet on Monterey, so it really should be skipping. Still working on the fix though.

**Comment by EricFromCanada** [comment]
> ```
> curl: (35) error:06FFF089:digital envelope routines:CRYPTO_internal:bad key length
> ```
Meanwhile on ARM: case sensitivity issue?
```
==> ./configure --disable-dependency-tracking --disable-silent-rules --prefix=/opt/homebrew/Cellar/neopop-sdl/0.2 --mandir=/opt/homebrew/Cellar/neopop-sdl/0.2/share/man
checking build system type... configure: error: /bin/sh ./config.sub -apple-darwin21.1.0 failed
rm: core: is a directory
```
And on Linux: may need `DISPLAY= ` in test
```
==> Testing neopop-sdl
==> /home/linuxbrew/.linuxbrew/Cellar/neopop-sdl/0.2/bin/NeoPop-SDL -V
cannot initialize SDL: No available video device
```

**Comment by EricFromCanada** [comment]
```
==> brew livecheck neopop-sdl --full-name --debug
==> FAILED
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/neopop-sdl.rb

Formula:          neopop-sdl
Livecheckable?:   Yes

URL (homepage):   https://nih.at/NeoPop-SDL/
Strategy:         PageMatch
Regex:            /href=.*?NeoPop-SDL[._-]v?(\d+(?:\.\d+)+)\.t/i
curl: (35) error:06FFF089:digital envelope routines:CRYPTO_internal:bad key length
Error: neopop-sdl: Unable to get versions
```
It seems livecheck isn't noticing the `:homebrew_curl` tag.

**Comment by carlocab** [comment]
Yep, that's https://github.com/Homebrew/brew/issues/12341

File: Formula/neopop-sdl.rb [hunk 1]
```diff
@@ -42,6 +43,11 @@ def install
   end
 
   test do
+    on_linux do
+      # Test fails on headless CI: "cannot initialize SDL: No available video device"
+      return if ENV["HOMEBREW_GITHUB_ACTIONS"]
+    end
```
[the remark this answers is not in the document]
Comment (EricFromCanada) [reply Formula/neopop-sdl.rb:49]: There's another 30 formulae that use an `on_linux do` block for this situation (e.g. `glew`, `tcl-tk`) so just confirming that this is the preferred style?

[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/neopop-sdl.rb:49]: It is. We even have [an `audit`](https://github.com/Homebrew/brew/blob/d9b7c207532263d8137a832a484c28bf530d86cb/Library/Homebrew/rubocops/lines.rb#L355-L357) for it; it just isn't working correctly.

**Review by carlocab — APPROVED** [review]
Ok with merging this despite failures.

**Pull request merged**